kubernetes/kubernetes_service.go

```diff
@@ -101,7 +103,13 @@ func (in *IstioClient) GetService(namespace, serviceName string) (*v1.Service, e
 // GetPods returns the pods definitions for a given set of labels.
 // It returns an error on any problem.
 func (in *IstioClient) GetPods(namespace string, labelsSet labels.Set) (*v1.PodList, error) {
-	formatted := labels.FormatLabels(labelsSet)
+	formatted := labelsSet.String()
+	// An empty selector is ambiguous in the go client, could mean either "select all" or "select none"
+	// Here we assume empty == select none; driven by use-case of namespace "default" / service "kubernetes" which has no pod(?)
+	// (see also https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors)
+	if formatted == "" {
+		return &v1.PodList{Items: []v1.Pod{}}, nil
```

Comment:
I agree that the client is a bit ambiguous, but I think it doesn't help if we have this exception here. I can imagine people sending empty label sets for getting all the deploymens/autoscalers/services/etc but having to change mind for fetching pods. IMO, It would be error-prone to follow the k8s-way for pods also.

Another point is that it is being returned empty metadata within that PodList. So this can be somehow problematic although it is been saved one API call.

Reply:
Just my 2 cents but it might be more explicit to check the label selector in `GetServiceDetails()` and don't call `GetPods()` if it is empty?

```go
   if len(service.Spec.Selector) > 0 {
 	go func() {
 		pods, err := in.GetPods(namespace, service.Spec.Selector)
 		podsChan <- podsResponse{pods: pods, err: err}
 	}()
   else {
      close(podsChan)
   }
```

Reply:
@simonpasquier i'm not sure, this is perfectly valid to have an empty selector, so when calling any api function like `GetPods` that takes a selector as parameter, as the caller I would assume it can receive any selector.

Reply:
@xeviknal I feel like both solutions are imperfect. But I would agree that the one you suggest is probably less imperfect :) I'll change that.

Reply:
@jotak Perhaps I lost something, but I guess that if we got a nil from param labelsSet, then we should send to `List` func a `<none> value. Otherwise, we just pass the labelsSet to `List`.
Does it have implications? (besides update `GetPods` callers)